.github/workflows/ci.yaml

```diff
@@ -6,7 +6,8 @@ on:
     - 'main'
     - '!dependabot/**'
     tags:
-    - '**'
+    # semver tags
+    - 'v[0-9]+.[0-9]+.[0-9]+-?**'
```

Comment:
Period characters in a regular expression can match any character.  You should prefix these periods with a backslash (i.e.: `\.`) to indicate that you really do want periods to separate the components in the semver string.